src/InvalidRequestException.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Amp\Http\Client;
+
+final class InvalidRequestException extends \Exception
```

Comment:
This should still extend the base exception, so `HttpException`.

Reply:
I didn't extend `HttpException` so that `RetryRequests` would not catch the exception, but instead I can catch and immediately re-throw.